docs/references/api/domain_representations.rst

```diff
@@ -0,0 +1,181 @@
+.. _domain_reps:
+
+Domain Representations
+######################
+
+Domain Representations separates "how the data is presented" from "how the data is stored". It works by creating `domains <https://www.postgresql.org/docs/current/sql-createdomain.html>`_ and `casts <https://www.postgresql.org/docs/current/sql-createcast.html>`_, the latter act on the former to present and receive the data in different formats.
```

Comment:
Nicely done! I struggled with coming up with a concise way to describe the feature.

Reply:
Thanks! Actually I stole some phrases from your [opening issue](https://github.com/PostgREST/postgrest/issues/2310#issue-1267408755) :smiley: 

I wanted to remove the "latter/former" words but I found it hard. Although it seems ok leaving it this way.

I've just addressed your latest reviews. LMK if you approve and I'll merge :rocket:.

Reply:
Also clarified the advantages (focusing on views) and "why not create a type?".